gammapy/estimators/utils.py

```diff
@@ -157,3 +155,72 @@ def estimate_exposure_reco_energy(dataset, spectral_model=None, normalize=True):
         reco_exposure = reco_exposure / ref_flux[:, np.newaxis, np.newaxis]
 
     return reco_exposure
+
+
+def _satisfies_conditions(info_dict, conditions):
+    satisfies = True
+    for key in conditions.keys():
+        satisfies &= info_dict[key.strip("_min")] > conditions[key]
+    return satisfies
+
+
+def resample_energy_edges(dataset, conditions={}):
+    """Returns energy edges that defining a binning that satisfies given conditions on the per-bin statistics.
+
+    Parameters
+    ----------
+    dataset:`~gammapy.datasets.SpectrumDataset` or `~gammapy.datasets.SpectrumDatasetOnOff`
+            the input dataset
+    conditions : dict
+        Keyword arguments containing the per-bin conditions used to resample the axis.
+        Available options are: 'counts_min', 'background_min', 'excess_min', 'sqrt_ts_min', 'npred_min',
+        'npred_background_min', 'npred_signal_min'.
+    Returns
+    -------
+    energy_edges : list of `~astropy.units.Quantity`
+        Energy edges for the resampled energy axis.
+    """
+    if not isinstance(dataset, SpectrumDataset) or isinstance(
```

Comment:
Minor: this could be written shorter as `isinstance(dataset, (SpectrumDataset, SpectrumDatasetOnOff))`